iocore/net/SSLNetVConnection.cc

```diff
@@ -303,6 +310,10 @@ ssl_read_from_net(SSLNetVConnection *sslvc, EThread *lthread, int64_t &ret)
     ret = bytes_read;
 
     event = (s->vio.ntodo() <= 0) ? SSL_READ_COMPLETE : SSL_READ_READY;
+    if (sslErr == SSL_ERROR_NONE && s->vio.ntodo() > 0) {
+      // We stopped with data on the wire (to avoid overbuffering).  Make sure we are triggered
+      sslvc->read.triggered = 1;
```

Comment:
Yes, we should return SSL_READ_READY.  Will fix that.  However, we should also set read trigger.  Otherwise, we will be waiting until the next read event.  If there is read data left, we will not get another read event on the epoll.